### PR TITLE
WIP: Migrate to poetry packaging system + Update build system

### PR DESCRIPTION
**Pull request by Granitosaurus**
Migrating to poetry and updating tox

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+[tool.poetry]
+name = "pyppeteer2"
+version = "0.2.2"
+description = "Headless chrome/chromium automation library (unofficial port of puppeteer)"
+readme = 'README.md'
+license = "MIT"
+homepage = "https://github.com/pyppeteer/pyppeteer2"
```
Comment (Mattwmaster58) [inline pyproject.toml:7]: Maybe the homepage should be the docs?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "Hiroyuki Takagi <miyako.dev@gmail.com>",
+]
+classifiers = [
+        'Development Status :: 3 - Alpha',
+        'Intended Audience :: Developers',
+        'License :: OSI Approved :: MIT License',
+        'Natural Language :: English',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
```
Comment (Mattwmaster58) [inline pyproject.toml:22]: I think we could safely add 3.8 in here

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+]
+
+[tool.poetry.urls]
+"Bug Tracker" = "https://github.com/pyppeteer/pyppeteer2/issues"
+
+[tool.poetry.scripts]
+pyppeteer-install = 'pyppeteer.command:install'
+
+[tool.poetry.dependencies]
+python = "^3.6.1"
```
Comment (Mattwmaster58) [inline pyproject.toml:48]: <s>I don't see any reason to restrict to < 3.6.1</s> I now see websockets requires it

File: tox.ini [hunk 4]
```diff
@@ -1,24 +1,21 @@
 [tox]
-envlist = py{35,36,37},doit
+envlist = py{38},doit
```
Comment (Mattwmaster58) [inline tox.ini]: Shouldn't we still include `py{36, 37}`?

File: tox.ini [hunk 5]
```diff
@@ -1,24 +1,21 @@
...
+  poetry
 commands =
+  poetry install -vv
   pyppeteer-install
-  py35: cp -r {toxinidir}/tests {toxworkdir}
-  py35: python -m unittest discover {toxworkdir}
   py36: python -m unittest discover
-  py37: coverage run -m unittest discover
-  py37: coverage report
+  py{37,38}: coverage run -m unittest discover
```
Comment (Mattwmaster58) [inline tox.ini]: Shouldn't we include `py{36}`?

File: pyproject.toml (hunk 2 above)
Comment (Granitosaurus) [reply pyproject.toml:22]: Done

File: pyproject.toml (hunk 1 above)
Comment (Granitosaurus) [reply pyproject.toml:7]: We kinda haven't migrated docs yet :grimacing: 
But in the future sure, that would probably be better

Comment (Mattwmaster58) [reply pyproject.toml:7]: Alright, leave it for now

File: README.md [hunk 6]
```diff
@@ -7,7 +7,7 @@ Pyppeteer2
 [![Travis status](https://travis-ci.org/miyakogi/pyppeteer2.svg)](https://travis-ci.org/miyakogi/pyppeteer2)
 [![AppVeyor status](https://ci.appveyor.com/api/projects/status/nb53tkg9po8v1blk?svg=true)](https://ci.appveyor.com/project/miyakogi/pyppeteer2)
 [![codecov](https://codecov.io/gh/miyakogi/pyppeteer2/branch/master/graph/badge.svg)](https://codecov.io/gh/miyakogi/pyppeteer2)
-_Note: this is a WIP continuation of pyppeteer project_  
+_Note: this is a WIP continuation of pyppeteer project_
```
Comment (Mattwmaster58) [inline README.md]:
```suggestion
_Note: this is a WIP continuation of pyppeteer project_ 
```

File: README.md [hunk 7]
```diff
@@ -7,7 +7,7 @@ Pyppeteer2
...
```
Comment (Mattwmaster58) [inline README.md]:
```suggestion
_Note: this is a WIP continuation of pyppeteer project_  
```

**Comment by Mattwmaster58** [comment]
Ugg, I merged my poetry branch into my dev branch so that CircleCI would build it...and promptly forgot about that when merging my dev with this poetry branch. Apologies for the mess.

Once this is merged, we'll still need to integrate with CircleCI (ie setup the project on the platform)

The commits I added mainly do the following:
 - use poetry in tox
 - use tox in circleci config
 - inline mypy.ini config -> now in tox.ini along with everything else
 - update poetry.lock

**Comment by Granitosaurus** [comment]
Github seems to find conflict in deletion of `setup.py` so this needs to be manually merged. Looks good to me!

**Comment by Mattwmaster58** [comment]
One of the things this will enable is nicely formatted test output with JUnit XML, both for mypy and tests.  The problem is I'd like to do everything through tox so that local tox result is the same as the remote one, and locally it doesn't really make sense for JUnit XML test results. 

I'll try and push the final commits out in the next day or so and then merge.

<details> <summary>Here's an example of the nicely formatted test output</summary>

screenshot

</details>

**Pull request merged**